src/engine/SCons/Tool/yacc.py

```diff
@@ -100,11 +100,15 @@ def yyEmitter(target, source, env):
 
 def get_yacc_path(env, append_paths=False):
     """
-    Find the a path containing the lex or flex binaries. If a construction 
-    environment is passed in then append the path to the ENV PATH.
+    Find the path to the yacc tool, searching several possible names
+
+    Only called in the Windows case, so the default_path
+    can be Windows-specific
+
+    :param env: current construction environment
+    :param append_paths: if set, add the path to the tool to PATH
+    :return: path to yacc tool, if found
     """
-    # save existing path to reset if we don't want to append any paths
-    envPath = env['ENV']['PATH']
     bins = ['bison', 'yacc', 'win_bison']
```

Comment:
Can you move this list to the top, name win_bins and have a non-win bins and then use that everywhere instead of having several lists scattered throughout the module?

Reply:
no problem, I've got that ready to push... but while I'm at it, the block in `yacc.py` from lines 145-151 seems like it is redundant now - like the "old version" was still left around when the `get_yacc_path` function was added. Should this be dropped?

Reply:
Sure.